spec/index.md

```diff
@@ -1954,6 +1954,15 @@ The optional `settings` parameter can be used to set the following settings:
 
     Default value: 5_000_000_000_000 (5 trillion cycles).
 
+-   `wasm_memory_limit` (`nat`)
+
+    Must be a number between 0 and 2<sup>48</sup>-1 (i.e., 256TB), inclusively, and indicates the upper limit on the WASM heap memory consumption of the canister.
+
+    An operation (update method, global timer, heartbeat, canister init, canister post_upgrade) that causes the WASM heap memory consumption to exceed this limit will trap.
+    The WASM heap memory limit is ignored for query methods, response callback handlers, and canister pre_upgrade.
+
+    Default value: 3_221_225_472 (3 GiB).
```

Comment:
There doesn't seem to be a default value in the implementation and an unspecified limit is returned as 0 in the canister status.

Reply:
The default is not set yet. The will be the final MR on the implementation side.